modules/bitgo/test/v2/unit/tss/ecdsa.ts

```diff
@@ -0,0 +1,242 @@
+        recipientGpgPublicArmor: bitgoGpgKeypair.publicKey,
+      }).should.be.rejectedWith('Invalid recipient');
+    });
+  });
+
+  describe('createCombinedKey', function () {
+    it('should create combined user key ecdsa', async function () {
+      const bitgoToUserShare = await encryptNShare({
+        keyShare: bitgoKeyShare,
+        recipientIndex: 1,
+        recipientGpgPublicArmor: userGpgKeypair.publicKey,
+        senderGpgPrivateArmor: bitgoGpgKeypair.privateKey,
+      });
+      const backupToUserShare = await encryptNShare({
+        keyShare: backupKeyShare,
+        recipientIndex: 1,
+        recipientGpgPublicArmor: userGpgKeypair.publicKey,
+        senderGpgPrivateArmor: backupGpgKeypair.privateKey,
+      });
+      const combinedUserKey = await createCombinedKey({
+        keyShare: userKeyShare,
+        commonKeychain,
+        encryptedNShares: [{
+          nShare: bitgoToUserShare,
+          recipientPrivateArmor: userGpgKeypair.privateKey,
+          senderPublicArmor: bitgoGpgKeypair.publicKey,
+        }, {
+          nShare: backupToUserShare,
+          recipientPrivateArmor: userGpgKeypair.privateKey,
+          senderPublicArmor: backupGpgKeypair.publicKey,
+        }],
+      });
+
+      combinedUserKey.commonKeychain.should.equal(commonKeychain);
+      combinedUserKey.signingMaterial.pShare.should.deepEqual(userKeyShare.pShare);
+      combinedUserKey.signingMaterial.backupNShare!.should.deepEqual(backupKeyShare.nShares[1]);
+      combinedUserKey.signingMaterial.bitgoNShare.should.deepEqual(bitgoKeyShare.nShares[1]);
+      should.not.exist(combinedUserKey.signingMaterial.userNShare);
+    });
+
+    it('should create combined backup key', async function () {
+      const bitgoToBackupShare = await encryptNShare({
+        keyShare: bitgoKeyShare,
+        recipientIndex: 2,
+        recipientGpgPublicArmor: backupGpgKeypair.publicKey,
+        senderGpgPrivateArmor: bitgoGpgKeypair.privateKey,
+      });
+
+      const userToBackupShare = await encryptNShare({
+        keyShare: userKeyShare,
+        recipientIndex: 2,
+        recipientGpgPublicArmor: backupGpgKeypair.publicKey,
+        senderGpgPrivateArmor: userGpgKeypair.privateKey,
+      });
+
+      const combinedBackupKey = await createCombinedKey({
+        keyShare: backupKeyShare,
+        commonKeychain,
+        encryptedNShares: [{
+          nShare: bitgoToBackupShare,
+          recipientPrivateArmor: backupGpgKeypair.privateKey,
+          senderPublicArmor: bitgoGpgKeypair.publicKey,
+        }, {
+          nShare: userToBackupShare,
+          recipientPrivateArmor: backupGpgKeypair.privateKey,
+          senderPublicArmor: userGpgKeypair.publicKey,
+        }],
+      });
+
+      combinedBackupKey.commonKeychain.should.equal(commonKeychain);
+      combinedBackupKey.signingMaterial.pShare.should.deepEqual(backupKeyShare.pShare);
+      combinedBackupKey.signingMaterial.userNShare!.should.deepEqual(userKeyShare.nShares[2]);
```

Comment:
is the` !` in `.signingMaterial.userNShare!`. intended ?

Reply:
yep